### PR TITLE
Dont initialize GS Aggregate Status until we process a GS

### PR DESCRIPTION
**Pull request by larohra**
## Description
Dont initialize GS Aggregate Status until we process a GS

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by larohra** [comment]
DCR run (with known setup issues) - https://tuxgold.corp.microsoft.com/job/AzLinux/job/DungeonCrawler/job/DungeonCrawler.larohra2/818/artifact/results/

File: azurelinuxagent/ga/exthandlers.py [hunk 1]
```diff
@@ -297,8 +292,10 @@ def _incarnation_changed(self, etag):
         return self.last_etag != etag
 
     def __last_gs_unsupported(self):
+
         # Return if the last GoalState was unsupported
-        return self.__gs_aggregate_status.status == GoalStateStatus.Failed and \
+        return self.__gs_aggregate_status is not None and \
```
Comment (narrieta) [inline azurelinuxagent/ga/exthandlers.py:297]: consider raising when it is None

**Review by narrieta — APPROVED** [review]
Approved with 1 suggestion

File: azurelinuxagent/ga/exthandlers.py (hunk 1 above)
Comment (larohra) [reply azurelinuxagent/ga/exthandlers.py:297]: discussed offline, leaving it as is for now

**Pull request merged**